packages/infra/lib/ihe-stack.ts

```diff
@@ -107,24 +113,32 @@ export class IHEStack extends Stack {
 
     const lambdaLayers = setupLambdasLayers(this, true);
 
+    const posthogSecretName = props.config.analyticsSecretNames?.POST_HOG_API_KEY;
+
     const documentQueryLambda = this.setupDocumentQueryLambda(
       props,
       lambdaLayers,
       vpc,
+      secrets,
```

Comment:
A couple comments here:

1. Instead of sending secrets + specific secret name for lookup, can't we just pass the retrieved secret value?
2. It's getting complicated with passing 6-7 arguments to these functions. It'll be very easy to mess the order of the parameters here. Can we wrap the parameters in an object?

Reply:
> can't we just pass the retrieved secret value

No, secrets should not be stored on env vars.

> It's getting complicated with passing 6-7 arguments to these functions. It'll be very easy to mess the order of the parameters here. Can we wrap the parameters in an object?

😍